### DISCUSSION TITLE
Updating dependencies for NVDA 2023.1

### DISCUSSION THREAD

**Discussion by josephsl** [nvaccess/nvda#14402] [General]
Hi,

Given that 2023.1 is planned as a compatibility breaking release, let us update dependencies. Candidates include:

* py2exe < 0.13.0 (0.13 drops support for Python 3.7)
* Flake8
* Sphinx
* System test tools such as pyautogui and Robot framework

Notes:

* wxPython 4.2.x is ideal provided that a Win32 wheel is made available.
* Python upgrade is not covered in this discussion.
* Can we use freeze function provided by Py2exe 0.12 and later?
* Let us dedicate one issue/PR per dependency and reserve this discussion for overall coordination.

Thanks.

**Comment by Brian1Gaff** [comment]
I was wondering  if, to avoid the problems encountered when a breaking 
change was proposed last time if there might be some way that a sub variant 
of 2022.4, could have a function that could report the possible breaking of 
add ons ahead of time to allow users and add on writers to address the 
changes before it became an issue, or failing that some kind of stand alone 
software to do it.
When Windows 7 support is lost, I can see that this warning would be simple 
to implement, but not sure about add ons.
 Brian

-- 
***@***.***
Sent via blueyonder.(Virgin media)
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
----- Original Message ----- 
From: "Joseph Lee" ***@***.***>
To: "nvaccess/nvda" ***@***.***>
Cc: "Subscribed" ***@***.***>
Sent: Saturday, November 26, 2022 12:58 PM
Subject: [nvaccess/nvda] Updating dependencies for NVDA 2023.1 (Discussion 
#14402)
